main/timeseries.py

```diff
@@ -84,6 +84,44 @@ def get_effort_timeseries(start_date: datetime, end_date: datetime) -> pd.Series
     return timeseries
 
 
+def get_charged_effort_timeseries(
+    start_date: datetime, end_date: datetime
+) -> pd.Series[float]:
+    """Get the timeseries data for aggregated charged project effort (External funding).
+
+    Args:
+        start_date: datetime object representing the start of the plotting period
+        end_date: datetime object representing the end of the plotting period
+
+    Returns:
+        Pandas series of aggregated effort with date range as index.
+    """
+    dates = pd.bdate_range(
+        pd.Timestamp(start_date), pd.Timestamp(end_date), inclusive="left"
+    )
+    # filter Projects to ensure dates exist and overlap with timeseries dates
+    projects = list(
+        models.Project.objects.filter(
+            start_date__lt=end_date.date(),
+            end_date__gte=start_date.date(),
+            start_date__isnull=False,
+            end_date__isnull=False,
+        )
+    )
+    projects = [
+        project
+        for project in projects
+        if project.funding_source.filter(source="External").exists()
+    ]
```

Comment:
Have set `source="External"` assuming that "External" projects are charged. Am I correct in assuming that only "External" projects are charged and "Internal" are not? Or are there any "Internal" projects which are charged?